front/app/containers/Admin/projects/project/analysis/InputPreview/index.tsx

```diff
@@ -41,6 +49,34 @@ const InputListItem = () => {
   const showManageIdeaButton =
     analysis.data.attributes.participation_method === 'ideation' && phaseId;
 
+  const isSurveyAnalysis =
+    analysis.data.attributes.participation_method === 'native_survey';
+
+  const mainCustomFieldId =
+    analysis.data.relationships?.main_custom_field?.data?.id;
+
+  const additionalCustomFieldIds =
+    analysis.data.relationships?.additional_custom_fields?.data.map(
+      (field) => field.id
+    );
+
+  const allCustomFields = analysis.data.relationships.all_custom_fields.data;
+  const customFieldsInAnalysisIds =
+    [mainCustomFieldId, ...(additionalCustomFieldIds || [])] || [];
```

Comment:
What happens if there is no main field? Will it be filtered out from the list, or will the first element of the list be null?

Reply:
Yes, in that case, the first element in the array will be `undefined` but this is not an issue in this case :)